Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -1739,6 +1739,71 @@ bool is_non_manifold_vertex(typename boost::graph_traits<PolygonMesh>::vertex_de
   return false;
 }
 
+/// \ingroup PMP_repairing_grp
+/// collects the non-manifold vertices (if any) present in the mesh.
+///
+/// @tparam PolygonMesh a model of `HalfedgeListGraph`
+/// @tparam OutputIterator a model of `OutputIterator` with value type `boost::graph_traits<PolygonMesh>::vertex_descriptor`
+///
+/// @param pm a triangle mesh containing `v`
+/// @param out the output iterator that collects the non-manifold vertices
+///
+/// \sa `is_non_manifold_vertex()`
+/// \sa `duplicate_non_manifold_vertices()`
+///
+/// \return the output iterator.
+template <typename PolygonMesh, typename OutputIterator>
+OutputIterator non_manifold_vertices(const PolygonMesh& pm,
+                                     OutputIterator out)
+{
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor      vertex_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor    halfedge_descriptor;
+
+  // Non-manifoldness can appear either:
+  // - if 'pm' is pinched at a vertex. While traversing the incoming halfedges at this vertex,
+  //   we will meet strictly more than one border halfedge.
+  // - if there are multiple umbrellas around a vertex. In that case, we will find a non-visited
+  //   halfedge that has for target a vertex that is already visited.
+
+  std::set<vertex_descriptor> non_manifold_vertices;
+  boost::unordered_set<vertex_descriptor> vertices_visited;
+  boost::unordered_set<halfedge_descriptor> halfedges_handled;
```

Comment:
what about dynamic properties :)

Reply:
Right. Probably can update to dynamic pmaps at other places too.